src/oidcc_token.erl

```diff
@@ -232,7 +232,13 @@ has_other_audience(ClientId, Audience) when is_list(Audience) ->
 
 
 validate_signature(IdToken, Kid, PubKeys, ProviderId) ->
-    PubKey = get_needed_key(PubKeys, Kid),
+    PubKeys1 =
+        case PubKeys of
+            [] -> refetch_keys(ProviderId);
+            _ -> PubKeys
+        end,
+
```

Comment:
For some reason (I only tested this using Keycloak as the Provider), the PubKeys are empty when you enter this function. So what happens eventually is that your callback gets called with login_failed and reason is function_clause.

To give you a better clue, I traced the exception to be thrown in oidcc_cowboy:handle_return, specifically:

```
[{erljwt,jwt_check_signature,
         [<<"fWbyDJXsFN2rLnbejeFw7u5bjNiRh9vTabZsnW49MNv2oBNhVYLUMgRuyQ4ySXClUFKmUyNmHOXcHOTIQPpPsHaa8ptlwJC2WPZa3YY8f6dXAgLVbq1UMAMjc5hBIwpJ20q1vkfx2oeFwhGSNtOSGZRV1hFUp9d6JwEOMPGeO2dhMDwGLs5Zz_djh0BfPIH6y5m_LAR6LE8X-YhSjb2gMnujWX8_l3EmARKtlyNAlvy7XryBFq0XyV
y920x3XlCpwLgFFLo9WC13WsMNyIybSGixsDD9AJKQCSMJIf89YugHdnwmwgYDty0GdfkeI6JERLWFajPawGSiGFhHRkHu4w">>,
          <<"RS256">>,
          <<"eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJmakROdlJEdUg5YUw3MVpaVGFTUEVPRGF3OXpzR1FuSzJSdzJ5bXBjN0hNIn0.eyJqdGkiOiJiZGI4MWJhNi03NGY0LTRkN2ItYmExMS1jZjMyMTc1YjFkNzYiLCJleHAiOjE0ODcxNjY0OTMsIm5iZiI6MCwiaWF0IjoxNDg3MTY2NDMzLCJpc3MiOiJodHRwOi
8vbG9jYWxob3N0L2F1dGgvcmVhbG1zL21hc3RlciIsImF1ZCI6ImFsdHdvcngiLCJzdWIiOiIwMzZiMGQzMC0xODc1LTQyYzMtOGI2OS1lZGNkNGI4ZDg3YjkiLCJ0eXAiOiJJRCIsImF6cCI6ImFsdHdvcngiLCJub25jZSI6ImpSeF9WQm13cTVNY0huYlJnQlRmbDNkZ0ZhTG9UZjJDUUtsZmFjVEhKcG1ENmtkLWxfdWticEcxTmJ4OUdfX0NxT
nNvU01GdWJuWkl0NFV1cjQxUERYdXZYV0FxTC1LSVNlQmRrWnhsck80aDRTdUFCOG4zQW5nRW5MaEsyUndMTUNJcjFPOE55LWRTZENjcFpEUm1BaDFsX3JETlkzNFBNcDl3cXE0VDJ5YyIsImF1dGhfdGltZSI6MTQ4NzE2NDczNiwic2Vzc2lvbl9zdGF0ZSI6IjViZmIwYWU5LWFiZTgtNDRjYS05ZGU0LTc0NjNkMjg2NmU0MiIsImFjciI6IjAi
LCJuYW1lIjoiRGV2IEFkbWluIiwicHJlZmVycmVkX3VzZXJuYW1lIjoiYXdhZG1pbiIsImdpdmVuX25hbWUiOiJEZXYiLCJmYW1pbHlfbmFtZSI6IkFkbWluIiwiZW1haWwiOiJhd2FkbWluQGFsdHdvcnguY29tIn0">>,
          no_key],
         [{file,"/Users/akovari/Projects/altworx/_build/default/lib/erljwt/src/erljwt.erl"},
          {line,148}]},
 {erljwt,parse_jwt,3,
         [{file,"/Users/akovari/Projects/altworx/_build/default/lib/erljwt/src/erljwt.erl"},
          {line,70}]},
 {oidcc_token,validate_signature,4,
              [{file,"/Users/akovari/Projects/altworx/_checkouts/oidcc/src/oidcc_token.erl"},
               {line,247}]},
 {oidcc_token,int_validate_id_token,4,
              [{file,"/Users/akovari/Projects/altworx/_checkouts/oidcc/src/oidcc_token.erl"},
               {line,161}]},
 {oidcc_token,validate_id_token,4,
              [{file,"/Users/akovari/Projects/altworx/_checkouts/oidcc/src/oidcc_token.erl"},
               {line,72}]},
 {oidcc_token,validate_id_token_map,4,
              [{file,"/Users/akovari/Projects/altworx/_checkouts/oidcc/src/oidcc_token.erl"},
               {line,62}]},
 {oidcc_token,validate_token_map,4,
              [{file,"/Users/akovari/Projects/altworx/_checkouts/oidcc/src/oidcc_token.erl"},
               {line,30}]},
 {oidcc_cowboy,handle_return,2,
               [{file,"/Users/akovari/Projects/altworx/_checkouts/oidcc_cowboy/src/oidcc_cowboy.erl"},
                {line,105}]}]
```